### ISSUE TITLE
StackOverflowException occurs sporadically when plugin enabled

### ISSUE THREAD

**Issue by dbeg** [mplushnikov/lombok-intellij-plugin#379]
IntelliJ IDEA 2016.3.6
Build #IU-163.15188.11, built on March 24, 2017
JRE: 1.8.0_112-release-408-b6 x86_64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o

```
null
java.lang.StackOverflowError
	at com.intellij.util.containers.hash.EqualityPolicy$2.isEqual(EqualityPolicy.java:44)
	at com.intellij.util.containers.hash.LinkedHashMap.get(LinkedHashMap.java:93)
	at com.intellij.util.containers.SLRUMap.get(SLRUMap.java:77)
	at com.intellij.util.containers.SLRUCache.get(SLRUCache.java:41)
	at com.intellij.util.indexing.MapIndexStorage.read(MapIndexStorage.java:422)
	at com.intellij.util.indexing.MapIndexStorage.read(MapIndexStorage.java:52)
	at com.intellij.util.indexing.MemoryIndexStorage.read(MemoryIndexStorage.java:214)
	at com.intellij.util.indexing.MapReduceIndex.getData(MapReduceIndex.java:395)
	at com.intellij.psi.stubs.StubIndexImpl.a(StubIndexImpl.java:359)
	at com.intellij.psi.stubs.StubIndexImpl.processElements(StubIndexImpl.java:325)
	at com.intellij.psi.stubs.StubIndex.getElements(StubIndex.java:146)
	at com.intellij.psi.stubs.StubIndex.getElements(StubIndex.java:135)
	at com.intellij.psi.impl.java.stubs.index.JavaFullClassNameIndex.get(JavaFullClassNameIndex.java:48)
	at com.intellij.psi.impl.file.impl.JavaFileManagerImpl.doFindClasses(JavaFileManagerImpl.java:96)
	at com.intellij.psi.impl.file.impl.JavaFileManagerImpl.findClass(JavaFileManagerImpl.java:131)
	at de.plushnikov.intellij.plugin.extension.LombokElementFinder.findClass(LombokElementFinder.java:41)
	at com.intellij.psi.impl.JavaPsiFacadeImpl.doFindClass(JavaPsiFacadeImpl.java:122)
	at com.intellij.psi.impl.JavaPsiFacadeImpl.findClass(JavaPsiFacadeImpl.java:99)
	at com.intellij.jsf.utils.JsfCommonUtils.lambda$isFlowSupported$0(JsfCommonUtils.java:102)
	at com.intellij.openapi.project.DumbService$1.run(DumbService.java:94)
	at com.intellij.openapi.project.DumbService.runReadActionInSmartMode(DumbService.java:124)
	at com.intellij.openapi.project.DumbService.runReadActionInSmartMode(DumbService.java:91)
	at com.intellij.jsf.utils.JsfCommonUtils.isFlowSupported(JsfCommonUtils.java:100)
	at com.intellij.jsf.utils.JsfCommonUtils.isJsf_2_2(JsfCommonUtils.java:92)
	at com.intellij.jsf.schemas.JsfSchemaProvider_JavaEE7.isAvailable(JsfSchemaProvider_JavaEE7.java:39)
	at com.intellij.xml.XmlSchemaProvider.findSchema(XmlSchemaProvider.java:52)
	at com.intellij.xml.XmlSchemaProvider.findSchema(XmlSchemaProvider.java:67)
	at com.intellij.xml.util.XmlUtil.findNamespace(XmlUtil.java:240)
	at com.intellij.psi.impl.source.xml.XmlDocumentImpl.getDefaultNSDescriptorInner(XmlDocumentImpl.java:255)
	at com.intellij.psi.impl.source.xml.XmlDocumentImpl.lambda$getDefaultNSDescriptor$0(XmlDocumentImpl.java:189)
	at com.intellij.psi.impl.PsiCachedValueImpl.doCompute(PsiCachedValueImpl.java:49)
	at com.intellij.util.CachedValueBase.getValueWithLock(CachedValueBase.java:222)
	at com.intellij.psi.impl.PsiCachedValue.getValueWithLock(PsiCachedValue.java:65)
	at com.intellij.psi.impl.PsiCachedValueImpl.getValue(PsiCachedValueImpl.java:38)
	at com.intellij.psi.impl.source.xml.XmlDocumentImpl.getDefaultNSDescriptor(XmlDocumentImpl.java:200)
	at com.intellij.psi.impl.source.xml.XmlDocumentImpl.getDefaultNSDescriptorInner(XmlDocumentImpl.java:247)
	at com.intellij.psi.impl.source.xml.XmlDocumentImpl.lambda$getDefaultNSDescriptor$0(XmlDocumentImpl.java:189)
	at com.intellij.psi.impl.PsiCachedValueImpl.doCompute(PsiCachedValueImpl.java:49)
	at com.intellij.util.CachedValueBase.getValueWithLock(CachedValueBase.java:222)
	at com.intellij.psi.impl.PsiCachedValue.getValueWithLock(PsiCachedValue.java:65)
	at com.intellij.psi.impl.PsiCachedValueImpl.getValue(PsiCachedValueImpl.java:38)
	at com.intellij.psi.impl.source.xml.XmlDocumentImpl.getDefaultNSDescriptor(XmlDocumentImpl.java:200)
	at com.intellij.tapestry.intellij.lang.descriptor.DescriptorUtil.getHtmlNSDescriptor(DescriptorUtil.java:250)
	at com.intellij.tapestry.intellij.lang.descriptor.TapestryXmlExtension.getDescriptorFromDoctype(TapestryXmlExtension.java:122)
	at com.intellij.psi.impl.source.xml.XmlDocumentImpl.getDefaultNSDescriptorInner(XmlDocumentImpl.java:273)
	at com.intellij.psi.impl.source.xml.XmlDocumentImpl.lambda$getDefaultNSDescriptor$0(XmlDocumentImpl.java:189)
	at com.intellij.psi.impl.PsiCachedValueImpl.doCompute(PsiCachedValueImpl.java:49)
	at com.intellij.util.CachedValueBase.getValueWithLock(CachedValueBase.java:222)
	at com.intellij.psi.impl.PsiCachedValue.getValueWithLock(PsiCachedValue.java:65)
	at com.intellij.psi.impl.PsiCachedValueImpl.getValue(PsiCachedValueImpl.java:38)
	at com.intellij.psi.impl.source.xml.XmlDocumentImpl.getDefaultNSDescriptor(XmlDocumentImpl.java:200)
	at com.intellij.psi.impl.source.xml.XmlDocumentImpl.getDefaultNSDescriptorInner(XmlDocumentImpl.java:247)
	at com.intellij.psi.impl.source.xml.XmlDocumentImpl.lambda$getDefaultNSDescriptor$0(XmlDocumentImpl.java:189)
	at com.intellij.psi.impl.PsiCachedValueImpl.doCompute(PsiCachedValueImpl.java:49)
	at com.intellij.util.CachedValueBase.getValueWithLock(CachedValueBase.java:222)
	at com.intellij.psi.impl.PsiCachedValue.getValueWithLock(PsiCachedValue.java:65)
	at com.intellij.psi.impl.PsiCachedValueImpl.getValue(PsiCachedValueImpl.java:38)
	at com.intellij.psi.impl.source.xml.XmlDocumentImpl.getDefaultNSDescriptor(XmlDocumentImpl.java:200)
	at com.intellij.tapestry.intellij.lang.descriptor.DescriptorUtil.getHtmlNSDescriptor(DescriptorUtil.java:250)
	at com.intellij.tapestry.intellij.lang.descriptor.TapestryXmlExtension.getDescriptorFromDoctype(TapestryXmlExtension.java:122)
	at com.intellij.psi.impl.source.xml.XmlDocumentImpl.getDefaultNSDescriptorInner(XmlDocumentImpl.java:273)
	at com.intellij.psi.impl.source.xml.XmlDocumentImpl.lambda$getDefaultNSDescriptor$0(XmlDocumentImpl.java:189)
	at com.intellij.psi.impl.PsiCachedValueImpl.doCompute(PsiCachedValueImpl.java:49)
	at com.intellij.util.CachedValueBase.getValueWithLock(CachedValueBase.java:222)
	at com.intellij.psi.impl.PsiCachedValue.getValueWithLock(PsiCachedValue.java:65)
	at com.intellij.psi.impl.PsiCachedValueImpl.getValue(PsiCachedValueImpl.java:38)
	at com.intellij.psi.impl.source.xml.XmlDocumentImpl.getDefaultNSDescriptor(XmlDocumentImpl.java:200)
	at com.intellij.psi.impl.source.xml.XmlDocumentImpl.getDefaultNSDescriptorInner(XmlDocumentImpl.java:247)
	at com.intellij.psi.impl.source.xml.XmlDocumentImpl.lambda$getDefaultNSDescriptor$0(XmlDocumentImpl.java:189)
	at com.intellij.psi.impl.PsiCachedValueImpl.doCompute(PsiCachedValueImpl.java:49)
	at com.intellij.util.CachedValueBase.getValueWithLock(CachedValueBase.java:222)
	at com.intellij.psi.impl.PsiCachedValue.getValueWithLock(PsiCachedValue.java:65)
	at com.intellij.psi.impl.PsiCachedValueImpl.getValue(PsiCachedValueImpl.java:38)
	at com.intellij.psi.impl.source.xml.XmlDocumentImpl.getDefaultNSDescriptor(XmlDocumentImpl.java:200)
	at com.intellij.tapestry.intellij.lang.descriptor.DescriptorUtil.getHtmlNSDescriptor(DescriptorUtil.java:250)
	at com.intellij.tapestry.intellij.lang.descriptor.TapestryXmlExtension.getDescriptorFromDoctype(TapestryXmlExtension.java:122)
	at com.intellij.psi.impl.source.xml.XmlDocumentImpl.getDefaultNSDescriptorInner(XmlDocumentImpl.java:273)
	at com.intellij.psi.impl.source.xml.XmlDocumentImpl.lambda$getDefaultNSDescriptor$0(XmlDocumentImpl.java:189)
	at com.intellij.psi.impl.PsiCachedValueImpl.doCompute(PsiCachedValueImpl.java:49)
	at com.intellij.util.CachedValueBase.getValueWithLock(CachedValueBase.java:222)
	at com.intellij.psi.impl.PsiCachedValue.getValueWithLock(PsiCachedValue.java:65)
	at com.intellij.psi.impl.PsiCachedValueImpl.getValue(PsiCachedValueImpl.java:38)
	at com.intellij.psi.impl.source.xml.XmlDocumentImpl.getDefaultNSDescriptor(XmlDocumentImpl.java:200)
	at com.intellij.psi.impl.source.xml.XmlDocumentImpl.getDefaultNSDescriptorInner(XmlDocumentImpl.java:247)
	at com.intellij.psi.impl.source.xml.XmlDocumentImpl.lambda$getDefaultNSDescriptor$0(XmlDocumentImpl.java:189)
	at com.intellij.psi.impl.PsiCachedValueImpl.doCompute(PsiCachedValueImpl.java:49)
	at com.intellij.util.CachedValueBase.getValueWithLock(CachedValueBase.java:222)
	at com.intellij.psi.impl.PsiCachedValue.getValueWithLock(PsiCachedValue.java:65)
	at com.intellij.psi.impl.PsiCachedValueImpl.getValue(PsiCachedValueImpl.java:38)
	at com.intellij.psi.impl.source.xml.XmlDocumentImpl.getDefaultNSDescriptor(XmlDocumentImpl.java:200)
	at com.intellij.tapestry.intellij.lang.descriptor.DescriptorUtil.getHtmlNSDescriptor(DescriptorUtil.java:250)
	at com.intellij.tapestry.intellij.lang.descriptor.TapestryXmlExtension.getDescriptorFromDoctype(TapestryXmlExtension.java:122)
	at com.intellij.psi.impl.source.xml.XmlDocumentImpl.getDefaultNSDescriptorInner(XmlDocumentImpl.java:273)
	at com.intellij.psi.impl.source.xml.XmlDocumentImpl.lambda$getDefaultNSDescriptor$0(XmlDocumentImpl.java:189)
	...
```

**Comment by AlexejK** [comment]
Any sample projects that you can provide where this happens?
Just because it shows lombok in the stacktrace doesnt actually mean its necessary the plugins fault.

I'm not getting these issues anywhere in my projects (large and small). So would need a sample project to test on, including the `.idea` folder

**Comment by sescotti** [comment]
Hi, I could reproduce this with this simple case:

```
public class A {
@Delegate
private A collaborator;
}
```

It was a typo on my side but I came across with this issue before that I noticed it, and you may want to know this (anyway, the Lombok docs state that recursive delegation is not supported).
 
Here's the stacktrace:
https://gist.github.com/sescotti/e725d7325f1dcddf2e22131eae06b5cd

**Comment by mplushnikov** [comment]
@sescotti it looks like to be another problem, I will create new issue for it. Thank you!

**Comment by mplushnikov** [comment]
I'm still not sure, what is the cause of your exception @dbeg, but I just added some prevention for possible recursion call. It should be available in the next release of plugin (0.16)
